termwiz/src/cell.rs

```diff
@@ -881,17 +882,40 @@ impl Cell {
     }
 }
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq, FromDynamic, ToDynamic)]
+pub struct CellWidth {
+    pub first: u32,
+    pub last: u32,
+    pub width: u8,
+}
+
+pub fn setcellwidths(cellwidths: Option<Vec<CellWidth>>) -> Option<HashMap<u32, u8>> {
+    if let Some(ref cellwidths) = cellwidths {
+        let mut map: HashMap<u32, u8> = HashMap::new();
+        for cellwidth in cellwidths {
+            for i in cellwidth.first..cellwidth.last+1 {
+                map.insert(i, cellwidth.width);
+            }
+        }
+        return Some(map);
+    } else {
+        return None;
+    }
```

Comment:
FWIW, you can simply this, something like:

```suggestion
    let cellwidths = cellwidths.as_ref()?;
    let mut map = HashMap::new();
    for cellwidth in cellwidths {
        for i in cellwidth.first..=cellwidth.last {
            map.insert(i, cellwidth.width);
        }
    }
    Some(map)
```